### PR TITLE
MULE-9885: Interface can be used as the type of a @ParameterGroup yet…

### PR DESCRIPTION
**Pull request by pablomusumeci**
… no error or warning is thrown

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java [hunk 1]
```diff
@@ -443,6 +446,15 @@ private void declareParameterGroups(Class<?> annotatedType, ParameterizedDeclare
                                                                           annotatedType));
             }
 
+            if (!isInstantiable(field.getType(), false))
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java]: this is in the wrong place. This should be done in a ModelEnricher which uses an ExtensionWalker to verify all the parameters which have the proper model property. If you do it here, then you're not validating alternative describers which might have the same bug

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java [hunk 2]
```diff
@@ -117,36 +121,20 @@ public void visitObject(ObjectType objectType)
...
-            validateNameCollisionWithTypes(parameterModel, ownerName, ownerModelType, extensionName,
-                                           parameters.stream().map(p -> hyphenize(p.getName())).collect(toList()));
-        });
+        new ExtensionWalker()
+        {
+            @Override
+            public void onParameter(ParameterizedModel owner, ParameterModel model)
+            {
+                String ownerName = owner.getName();
+                String ownerModelType = owner.getClass().getSimpleName();
```
Comment (pablomusumeci) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java]: We used to have an specific tag (CONNECTION_PROVIDER, OPERATION, CONFIGURATION, etc) to use in the error message,  but now with the ExtensionWalker it is no longer possible to identify what kind of component is owner. Any idea?

Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java]: a simple instanceof on the owner will do

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java [hunk 3]
```diff
@@ -191,4 +181,50 @@ private void validateNameCollisionWithTypes(ParameterModel parameterModel, Strin
...
+
+        if (parameterGroup != null)
+        {
+            parameterGroup.getGroups().stream()
+                    .map(p -> p.getField())
+                    .forEach(field -> {
+                        if (!isInstantiable(field.getType(), false))
+                        {
+                            throw new IllegalParameterModelDefinitionException(
+                                    format("Parameter group type should be instantiable. Affected field [%s] in [%s].",
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java]: rephase this to something more understandable. Say something like `an instantiable type with a default constructor was expected, however type %s was expected`

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java [hunk 4]
```diff
@@ -191,4 +181,50 @@ private void validateNameCollisionWithTypes(ParameterModel parameterModel, Strin
...
+    private void validateParameterGroup(ParameterModel parameterModel)
+    {
+        ParameterGroupModelProperty parameterGroup = parameterModel
+                .getModelProperty(ParameterGroupModelProperty.class)
+                .orElse(null);
+
+        if (parameterGroup != null)
+        {
+            parameterGroup.getGroups().stream()
+                    .map(p -> p.getField())
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java]: you don't have to check the fields... you have to check the type

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java [hunk 5]
```diff
@@ -190,14 +190,14 @@ private void validateParameterGroup(ParameterModel parameterModel)
...
+                    .forEach(parameter -> {
+                        if (!isInstantiable(parameter.getType()))
                         {
                             throw new IllegalParameterModelDefinitionException(
-                                    format("Parameter group type should be instantiable. Affected field [%s] in [%s].",
-                                           field.getName(),
-                                           field.getAnnotatedType()));
+                                    format("ParameterGroup type should be non abstract with a default constructor, however type '%s' was received. Affected field [%s] in [%s].",
+                                           parameter.getType().getName(),
+                                           parameter.getField().getName(),
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java]: why do you insist with the field?? the field is an internal thing which has nothing to do with this. If the `@ParameterGroup` is on an operation, what does the field tell the user. Just use parameter.getType() and nothing else

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java [hunk 6]
```diff
@@ -191,4 +181,46 @@ private void validateNameCollisionWithTypes(ParameterModel parameterModel, Strin
 
     }
 
-}
+    private void validateParameterGroup(ParameterModel parameterModel, String ownerName, String ownerModelType, String extensionName)
+    {
+        ParameterGroupModelProperty parameterGroup = parameterModel
+                .getModelProperty(ParameterGroupModelProperty.class)
+                .orElse(null);
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java]: don't do this and check for null later, it's already an optional, go for `ifPresent`

**Pull request merged**